src/Tribe/Timezones.php

```diff
@@ -94,6 +94,36 @@ public static function wp_timezone_abbr( $date ) {
 	}
 
 	/**
+	 * Generates a timezone string from a UTC offset
+	 *
+	 * @param string $offset UTC offset
+	 *
+	 * @return string
+	 */
+	public static function generate_timezone_string_from_utc_offset( $offset ) {
+		if ( ! self::is_utc_offset( $offset ) ) {
+			return $offset;
+		}
+
+		// ensure we have the minutes on the offset
+		if ( ! strpos( $offset, ':' ) ) {
+			$offset .= ':00';
+		}
+
+		$offset = str_replace( 'UTC', '', $offset );
+
+		list( $hours, $minutes ) = explode( ':', $offset );
+		$seconds = $hours * 60 * 60 + $minutes * 60;
+
+		$timezone = timezone_name_from_abbr( '', $seconds, 1 );
```

Comment:
I initially was mimicking this [stackoverflow answer](http://stackoverflow.com/a/11896631/124581) to ensure we didn't bump into [a bug with `timezone_name_from_abbr()`](https://bugs.php.net/bug.php?id=44780)...but I suppose that's less clear than what WooCommerce is doing to accomplish a similar goal. I went ahead and duplicated the relevant code from WooCommerce and called that out in the docblock.